src/Documents.ts

```diff
@@ -829,6 +746,128 @@ class Documents implements Pointer.Library, Pointer.Validator, ObjectSchema.Reso
 		return persistedDocument.getFragment( fragment.id );
 	}
 
+	private setDataRequest( uri:string, requestOptions:HTTP.Request.Options, asContainer:boolean ):string {
+		if( RDF.URI.Util.isRelative( uri ) && ! this.context ) throw new Errors.IllegalArgumentError( "This Documents instance doesn't support relative URIs." );
+
+		if ( this.context ) {
+			if ( this.context.auth.isAuthenticated() ) this.context.auth.addAuthentication( requestOptions );
+			uri = this.context.resolve( uri );
+		}
+
+		HTTP.Request.Util.setContentTypeHeader( "application/ld+json", requestOptions );
+		HTTP.Request.Util.setAcceptHeader( "application/ld+json", requestOptions );
+
+		let interactionModel:string = asContainer ? NS.LDP.Class.Container : NS.LDP.Class.RDFSource;
```

Comment:
The interaction model can be more things aside RDFSorce or Container. But all the requests need to specify their interaction model, so lets make the interaction model an argument of this method so each request specifies it.